src/presenced.cpp

```diff
@@ -350,6 +350,8 @@ Client::reconnect(const std::string& url)
                     mConnectPromise.reject(err.msg(), err.code(), WS_ERRTYPE_DNS);
                     mLoginPromise.reject(err.msg(), err.code(), WS_ERRTYPE_DNS);
                 }
+
+                return err;
```

Comment:
Look again at the code - the returned promise is not the chain output, it's mConnectPromise, so the chain output doesn't matter. If you insist to have the .fail() return the error, then you have to add another artificial .fail() which doesn't return an error. Otherwise you will get the "Unhandled promise fail" warning. Also, this code (as in chatd.cpp) silently ignores non-DNS errors, which is a major issue.